### DISCUSSION TITLE
LASSO Cox differences between packages

### DISCUSSION THREAD

**Discussion by alberto-mora** [sebp/scikit-survival#308] [Q&A]
I have a list of genes and I have performed a Penalized Cox Model LASSO (cox_lasso = CoxnetSurvivalAnalysis(l1_ratio=1.0, alpha_min_ratio=0.01) to select the most useful prognostic genes to be included later in a multivariate Cox regression to generate a risk score. 
I have also used the R package glmnet (with the parameter family = "cox"), to perform LASSO but the list of genes I have obtained is different from the one I have obtained from this package.
What are the differences between the two LASSO models? Which one is best for what I want to do?
Thanks in advance.

**Comment by sebp** [comment]
The implementation in scikit-survival is based on glmnet version 2. Back then, the only difference was that the solution computed by glmnet depended on the order of the samples (if there are ties), whereas the one in scikit-survival does not. Since the current version of glmnet is 4, there might be other differences, but I haven't checked their code since version 2, so can't tell you exactly.
